cdap-security/src/main/java/co/cask/cdap/security/authorization/AbstractAuthorizationService.java

```diff
@@ -176,6 +178,18 @@ protected void shutDown() throws Exception {
     return cacheEnabled ? authPolicyCache.get(principal) : fetchPrivileges(principal);
   }
 
+  protected void doInvalidate(final Predicate<Principal> predicate) {
+    Iterable<Principal> filtered = Iterables.filter(authPolicyCache.asMap().keySet(), new Predicate<Principal>() {
+      @Override
+      public boolean apply(Principal input) {
+        return predicate.apply(input);
```

Comment:
Why need to new a `Predicate` and just calling the other? Can't you just use the `predicate` in the argument?